src/sentry/options/manager.py

```diff
@@ -145,12 +140,16 @@ def get(self, key, silent=False):
         # Some values we don't want to allow them to be configured through
         # config files and should only exist in the datastore
         if opt.flags & FLAG_STOREONLY:
+            if callable(opt.default):
+                return opt.default()
             return opt.default
 
         try:
             # default to the hardcoded local configuration for this key
             return settings.SENTRY_OPTIONS[key]
         except KeyError:
+            if callable(opt.default):
+                return opt.default()
```

Comment:
It might be simpler to always make this a callable, either explicitly as a requirement of the `Option` constructor, or wrapping it there in a `lambda` if it's not already a callable.